types/node/globals.d.ts

```diff
@@ -187,59 +165,15 @@ declare function clearImmediate(immediateId: NodeJS.Immediate): void;
 
 declare function queueMicrotask(callback: () => void): void;
 
-// TODO: change to `type NodeRequireFunction = (id: string) => any;` in next mayor version.
-interface NodeRequireFunction {
-    /* tslint:disable-next-line:callable-types */
-    (id: string): any;
-}
-
-interface NodeRequire extends NodeRequireFunction {
-    resolve: RequireResolve;
-    cache: any;
-    /**
-     * @deprecated
-     */
-    extensions: NodeExtensions;
-    main: NodeModule | undefined;
-}
-
-interface RequireResolve {
-    (id: string, options?: { paths?: string[]; }): string;
-    paths(request: string): string[] | null;
-}
-
-interface NodeExtensions {
-    '.js': (m: NodeModule, filename: string) => any;
-    '.json': (m: NodeModule, filename: string) => any;
-    '.node': (m: NodeModule, filename: string) => any;
-    [ext: string]: (m: NodeModule, filename: string) => any;
-}
-
-declare var require: NodeRequire;
-
-interface NodeModule {
-    exports: any;
-    require: NodeRequireFunction;
-    id: string;
-    filename: string;
-    loaded: boolean;
-    parent: NodeModule | null;
-    children: NodeModule[];
-    paths: string[];
-}
-
-declare var module: NodeModule;
+declare var require: NodeJS.Require;
+declare var module: NodeJS.Module;
 
 // Same as module.exports
 declare var exports: any;
 
 // Buffer class
 type BufferEncoding = "ascii" | "utf8" | "utf-8" | "utf16le" | "ucs2" | "ucs-2" | "base64" | "latin1" | "binary" | "hex";
 
-interface Buffer {
```

Comment:
Why remove this?

Reply:
It's not really needed and was a special case that was added just because someone wanted to have access to the type safe constructor ¯\\_(ツ)_/¯

Reply:
I think I remember... was related to webpack to avoid that too much is packed.
So you plan is to break the build for this user? 😕

Reply:
That sounds odd, usually types don't affect webpack builds.

Reply:
see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/28900#issuecomment-421829307

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/28900 there is no mention of `Webpack` in this issue. Seems like it was pure _convenience_, since buffer was moved to a class it most likely doesn't matter anymore.

Reply:
The comment was `Because if i do that, bundlers include an entire Buffer polyfill`, I assumed the bundler was webpack but could be anything.

Reply:
webpack doesn't implicitly emit node polyfills anymore.
And in any case the person can just cast `Buffer.constructor` to `typeof Buffer` 😛